src/compiler/emitter.ts

```diff
@@ -4707,10 +4702,10 @@ namespace ts {
                             includeComments => getLinesBetweenPositionAndPrecedingNonWhitespaceCharacter(
                                 firstChild.pos,
                                 parentNode.pos,
-                                currentSourceFile!,
+                                Debug.checkDefined(currentSourceFile),

```

Comment:
I changed all of these callbacks to truly eliminate `currentSourceFile!`, but arguably they could have their `!` restored as they all immediately call the callback. But, it felt nice to not have any assertions.

Reply:
Yeah, I'll flip it back.